corehq/apps/userreports/sql.py

```diff
@@ -30,12 +31,18 @@ def save(self, doc):
             table = self.get_table()
             for indicator_row in indicator_rows:
                 with self.engine.begin() as connection:
-                    # delete all existing rows for this doc to ensure we aren't left with stale data
-                    delete = table.delete(table.c.doc_id == doc['_id'])
-                    connection.execute(delete)
-                    all_values = {i.column.id: i.value for i in indicator_row}
-                    insert = table.insert().values(**all_values)
-                    connection.execute(insert)
+                    try:
+                        # delete all existing rows for this doc to ensure we aren't left with stale data
+                        delete = table.delete(table.c.doc_id == doc['_id'])
+                        connection.execute(delete)
+                        all_values = {i.column.id: i.value for i in indicator_row}
+                        insert = table.insert().values(**all_values)
+                        connection.execute(insert)
+                    except IntegrityError:
+                        # Someone beat us to it. Concurrent inserts can happen
+                        # when a doc is processed by the celery rebuild task
+                        # at the same time as the pillow.
+                        pass
```

Comment:
Have narrowed the try block.

I'm not sure what you mean by 'put a where condition on the insert'. Also
this does use [connection pooling](
http://docs.sqlalchemy.org/en/rel_0_9/core/pooling.html#sqlalchemy.pool.QueuePool)
so there won't ever be more than 15 concurrent connections open at a time.